[ALL TESTS] Update (#6817)
mlflow/mlflow: CHANGELOG.md

# CHANGELOG

## 1.29.0 (2022-09-16)

MLflow 1.29.0 includes several major features and improvements

Features:

> 

Reviewer: Can we add one more item for `Pipelines` that was just cherry-picked?

```suggestion

- [Pipelines] Improve performance and fidelity of dataset profiling in the scikit-learn regression Pipeline (#6792 @sunishsheth2009) 
```

- [Pipelines] Add an `mlflow pipelines get-artifact` CLI for retrieving Pipeline artifacts (#6517, @prithvikannan)
- [Pipelines] Introduce an option for skipping dataset profiling to the scikit-learn regression Pipeline (#6456, @apurva-koti)
- [Pipelines / UI] Display an `mlflow pipelines` CLI command for reproducing a Pipeline run in the MLflow UI (#6376, @hubertzub-db)
- [Tracking] Automatically generate friendly names for Runs if not supplied by the user (#6736, @BenWilson2)
- [Tracking] Add `load_text()`, `load_image()` and `load_dict()` fluent APIs for convenient artifact loading (#6475, @subramaniam02)
- [Tracking] Add `creation_time` and `last_update_time` attributes to the Experiment class (#6756, @subramaniam02)

> - [Tracking] Add official MLflow Tracking Server DockerFiles to MLflow repository (#6731, @oojo12)

Reviewer: ```suggestion
- [Tracking] Add official MLflow Tracking Server Dockerfiles to the MLflow repository (#6731, @oojo12)
```

# CHANGELOG

> ## 1.29.0 (2022-09-16)

Reviewer: Can we make sure that all of the features / bug fixes with the same tag (e.g. `[Tracking]`) are located next to each other (i.e. don't interleave them) and that Pipelines is first, followed by Tracking, UI, Models, Model Registry, Scoring?

MLflow 1.29.0 includes several major features and improvements

Features:

> - [Tracking] Add support for load_text, load_image, load_dict to the fluent API (#6475, @subramaniam02)

Reviewer: ```suggestion
- [Tracking] Add `load_text()`, `load_image()`, and `load_dict()` fluent APIs for convenient artifact loading #6475, @subramaniam02)
```

> - [Tracking] Add attributes `creation_time` and `last_update_time` to the Experiment class (#6756, @subramaniam02)

Reviewer: ```suggestion
- [Tracking] Add  `creation_time` and `last_update_time` attributes to the Experiment class (#6756, @subramaniam02)
```

> - [Tracking] Add a default generated run name to started runs if not supplied by user (#6736, @BenWilson2)

Reviewer: ```suggestion
- [Tracking] Automatically generate friendly names for Runs if not supplied by the user (#6736, @BenWilson2)
```

@BenWilson2 Can we move this to the top of the features list? It's just that great!

> - [Tracking] Add `pos_label` to `mlflow.evaluate` to define the positive condition value in binary classification evaluations (#6696, @harupy)

Reviewer: ```suggestion
- [Tracking] Add `pos_label` support to `mlflow.evaluate()` for identifying the positive class (#6696, @harupy)
```

> - [Tracking] Update RunInfo, CreateRun and UpdateRun classes to accept `run_name` as a named attribute (#6769, @apurva-koti)

Reviewer: Can we remove this? We reverted it in `master` because the implementation is incomplete.

> - [Build / Docker] Add MLflow Tracking Server DockerFile and Docker compose files (#6731, @oojo12)

Reviewer: Can we recategorize this as [Tracking]?

```suggestion
- [Docker] Add official MLflow Tracking Server DockerFile and Docker compose files (#6731, @oojo12)
```

> - [Build / Docker] [CLI] Add new CLI command for generating Dockerfiles for models (#6591, @anuarkaliyev23)

Reviewer: ```suggestion
- [Models] Add new CLI command for generating Dockerfiles for model serving (#6591, @anuarkaliyev23)
```

> - [Build] Add Docker development scripts (#6465, @hubertzub-db)

Reviewer: Can we move this into "Small bug fixes and documentation updates" since it's a feature for platform contributors and not a platform feature for users?

- [UI] Make URLs clickable in the MLflow Tracking UI (#6526, @marijncv)
- [UI] Introduce support for csv data preview within the artifact viewer pane (#6567, @nnethery)

> - [Model Registry / Models] Add support for creating dependency-inclusive wheeled models (#6586, @arjundc-db)

Reviewer: ```suggestion
- [Model Registry / Models] Introduce `mlflow.models.add_libraries_to_model` API for adding libraries to an MLflow Model (#6586, @arjundc-db)
```

> - [Models] Add support for model validation (#6582, @jerrylian-db)

Reviewer: Can we put this at the top of the `[Models]` section?

```suggestion
- [Models] Add model validation support to `mlflow.evaluate()` (#6582, @jerrylian-db)
```

> - [Models] Add support for pyfunc encapsulated model method calls other than `.predict()` (#6631, @skylarbpayne)

Reviewer: ```suggestion
- [Models] Add `predict_proba()` support to the pyfunc representation of scikit-learn models (#6631, @skylarbpayne)
```

> - [Models] Add confs for metric prefix and metric dataset info to `mlflow.evaluate()`, rename `accuracy` to `accuracy_score` (#6593, @dbczumar)

Reviewer: ```suggestion
- [Models] Make the metric name prefix and dataset info configurable in `mlflow.evaluate()` (#6593, @dbczumar)
```

> - [Models] Add utility for signature validation of logged model to dataset (#6494, @serena-ruan)

Reviewer: ```suggestion
- [Models] Add utility for validating the compatibility of a dataset with a model signature (#6494, @serena-ruan)
```

> - [Models] Add support for Decimal object to Double cast in ML Flow (#6600, @shitaoli-db)

Reviewer: ```suggestion
- [Models] Add support for Decimal type inference to MLflow Model schemas (#6600, @shitaoli-db)
```

> - [Models] Introduce sample_weights arg to `mlflow.evaluate()` (#6806, @dbczumar)

Reviewer: ```suggestion
- [Models] Introduce `sample_weights` argument to `mlflow.evaluate()` (#6806, @dbczumar)
```

> - [Sagemaker] Add the ability to override variant name during Sagemaker deployment (#6486, @nfarley-soaren)

Reviewer: ```suggestion
- [Scoring] Add the ability to specify a `variant_name` during Sagemaker deployment (#6486, @nfarley-soaren)
```

> - [Sagemaker] Add data capture config to sagemaker (#6423, @jonwiggins)

Reviewer: ```suggestion
- [Sagemaker] Support specifying a `data_capture_config` during SageMaker deployment (#6423, @jonwiggins)
```

> - [Sagemaker] Move data capture config to support upcoming feature (#6621, @dbczumar)

Reviewer: Can we move this to small bug fixes / doc updates?

- [Scoring] Add `/health` endpoint to scoring server (#6574, @gabriel-milan)

> - [Java / Tracking] Add `searchExperiments` API to Java client, deprecate `listExperiments` (#6561, @dbczumar)

Reviewer: ```suggestion
- [Tracking] Add `searchExperiments` API to Java client, and deprecate `listExperiments` (#6561, @dbczumar)
```

> - [R / Tracking] Add mlflow_search_experiments API to R client (#6576, @dbczumar)

Reviewer: ```suggestion
- [Tracking] Add `mlflow_search_experiments` API to R client and deprecate `mlflow_list_experiments` (#6576, @dbczumar)
```

> - [Pipelines / CLI] Add support for `get_artifact` in the cli (#6517, @prithvikannan)

Reviewer: ```suggestion
- [Pipelines] Add an `mlflow pipelines get-artifact` CLI for retrieving Pipeline artifacts (#6517, @prithvikannan)
```

> - [Pipelines] Add Pipelines step option to skip step profiling (#6456, @apurva-koti)

Reviewer: ```suggestion
- [Pipelines] Introduce an option for skipping dataset profiling to the scikit-learn regression Pipeline (#6456, @apurva-koti)
```

> - [Pipelines / UI] CLI command suggestion after running the pipeline (#6376, @hubertzub-db)

Reviewer: ```suggestion
- [Pipelines] Display an `mlflow pipelines` CLI command for reproducing a Pipeline run in the MLflow UI (#6376, @hubertzub-db)
```

Bug fixes:

> - [Scoring] Fix schema issue with 1-D arrays for TF / Keras (#6796, @LiamConnell)

Reviewer: ```suggestion
- [Models] Fix an inference failure with 1-dimensional tensor inputs in TensorFlow and Keras (#6796, @LiamConnell)
```

> - [Build] Increase minimum python versions in dev setup script (#6727, @BenWilson2)

Reviewer: Can we move this to small bug fixes & doc updates?

> - [Models] Scope Databricks inferred pip requirements to modules (#6722, @dbczumar)

Reviewer: Can we move this to small bug fixes & doc updates?

> - [Models] Fix duplicate `_SPARK_MODEL_PATH_SUB` regression (#6683, @gwy1995)

Reviewer: ```suggestion
- [Models] Fix a regression in the directory path structure of logged Spark Models that occurred in MLflow 1.28.0 (#6683, @gwy1995)
```

> - [Models] Fix bug when reloading modules (#6647, @Jooakim)

Reviewer: ```suggestion
- [Models] No longer reload the `__main__` module when loading model code (#6647, @Jooakim)
```

> - [Tracking] Enforce idempotent behavior with make run & experiment deletion / restoration (#6641, @dbczumar)

Reviewer: ```suggestion
- [Tracking] Make Run and Experiment deletion and restoration idempotent (#6641, @dbczumar)
```

> - [Tracking] Remove the fixed width css for the experimentlist container (#6569, @sunishsheth2009)

Reviewer: ```suggestion
- [UI] Fix an alignment bug affecting the Experiments list in the MLflow UI  (#6569, @sunishsheth2009)
```

> - [Models] Fix issue with using mlflowdbfs if dbutils is unavailable (#6508, @dbczumar)

Reviewer: Can we move this to small bug fixes & doc updates?

> - [Artifacts] Add _download_file in hdfs_artifact_repo (#6482, @shidianshifen)

Reviewer: ```suggestion
- [Artifacts] Fix an `mlflow server` compatibility issue with HDFS when serving artifacts (#6482, @shidianshifen)
```

Documentation updates:

> - [Docs / Scoring] Deprecate mlflow.sagemaker.deploy() & mlflow.sagemaker.delete() (#6651, @dbczumar)

Reviewer: ```suggestion
- [Scoring] Deprecate `mlflow.sagemaker.deploy()` in favor of `SageMakerDeploymentClient.create()` (#6651, @dbczumar)
```

(no need to explicitly call out delete - should follow naturally)

> - [Docs] Add soft delete in (#6637, @ninabacc-db)

Reviewer: Can we move this to small bug fixes and documentation updates?

> - [Tracking] Stabilize SearchExperiments API (#6551, @dbczumar)

Reviewer: ```suggestion
- [Tracking] Mark the SearchExperiments API as stable (#6551, @dbczumar)
```

> - [Docs] Update documentation for model version `run_link` field (#6632, @arpitjasa-db)

Reviewer: Can we move this to small bug fixes & doc updates?

> - [Model Registry / Tracking] Deprecate ListExperiments, ListRegisteredModels, ListRunInfos APIs (#6550, @dbczumar)

Reviewer: ```suggestion
- [Tracking / Model Registry] Deprecate the ListExperiments, ListRegisteredModels, and `list_run_infos()` APIs (#6550, @dbczumar)
```
